File: app/src/main/java/com/example/olesya/boardgames/interfaces/ClientCallback.java
package com.example.olesya.boardgames.interfaces;

public interface ClientCallback {

    //получение карты от сервера
    void addCardCallback(String card);

    //выбор карты игроком по тематике
    //ходит игрок
    void onUserTurnEvent();

    //игрок закончил ход
    void onUserFinishTurnEvent(String card);

    //выбор карточки со стола
    void onUserChooseEvent(int playersNum);

    //ходит ведущий
    void onMainTurnEvent();

    //ведущий закончил ход
    void onMainFinishTurnEvent(String card);

    void onMainStopRoundEvent();

}

File: app/src/main/java/com/example/olesya/boardgames/interfaces/ClientCallback.kt
package com.example.olesya.boardgames.interfaces

interface ClientCallback {

    //получение карты от сервера
    fun addCardCallback(card: String)

    fun userPickingEnabled(enabled: Boolean)

    fun userChoosingEnabled(enabled: Boolean, fromNum: Int)

    fun showMessage(message: String)

    fun usernameChanged(username: String)
}